### ISSUE TITLE
duplicate raw_load_cham2 ?

### ISSUE THREAD

**Issue by andypicke** [OceanMixingGroup/mixingsoftware#2]
I just updated to new Matlab and got an error when trying to run chameleon processing for EQ14 (because my path ordering changed in new Matlab). It looks like there are two 'raw_load_cham2.m' files. One is in Chameleon2/Version2105/ , and the other is in /marlcham. It looks like the /marlcham one is older, although that one works for me and the other one gives an error (i'm running w/ the processing scripts Sally sent me, maybe they've been update since then).

**Comment by andypicke** [comment]
Btw, the error I got was: 

Field assignment to a non-structure array object.
Error in raw_load_cham2 (line 56)
  q.script.prefix='fake';

**Comment by sallywarner** [comment]
Hi Andy,

What version of matlab are you using? Are you using a mac or pc?

I am using matlabr2016a on a mac, and the raw_load_cham2.m in Chameleon/version2015 works for me with the eq14 data. 

I've had trouble with this same line of code when going between different machines that are using different versions of matlab. It's that damn global variable q!

I really think it's a matlab version issue, but I could be wrong...

Thanks for pointing this out, Andy. I went through mixingsoftware carefully, but obviously there will be things that are missed as people run the code on different machines.

thanks,
sally

**Comment by andypicke** [comment]
I got the error using matlab2016a on a mac. But it seems to run fine when I
use version 2015a. Actually I think I am calling the /marlcham/ version in
both. Should the Chameleon2/Version2015 folder be before /marlcham ?

On Wed, Jun 29, 2016 at 4:22 PM, sallywarner notifications@github.com
wrote:

> Hi Andy,
> 
> What version of matlab are you using? Are you using a mac or pc?
> 
> I am using matlabr2016a on a mac, and the raw_load_cham2.m in
> Chameleon/version2015 works for me with the eq14 data.
> 
> I've had trouble with this same line of code when going between different
> machines that are using different versions of matlab. It's that damn global
> variable q!
> 
> I really think it's a matlab version issue, but I could be wrong...
> 
> Thanks for pointing this out, Andy. I went through mixingsoftware
> carefully, but obviously there will be things that are missed as people run
> the code on different machines.
> 
> thanks,
> sally
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/OceanMixingGroup/mixingsoftware/issues/2#issuecomment-229507316,
> or mute the thread
> https://github.com/notifications/unsubscribe/AG6El47LBK65Y2mgHZ4Sckczc8CetKveks5qQvA0gaJpZM4JBnZ-
> .

## 

Andy Pickering
Postdoctoral Research Associate
Oregon State University
College of Earth, Ocean, and Atmospheric Sciences
978-764-3009
apickering@coas.oregonstate.edu
andypicke@gmail.com
Github https://github.com/andypicke

**Comment by sallywarner** [comment]
okay. I think that makes sense. The marlcham version worked with old matlabs, but in 2016a (and maybe 2015a) you can't write over a global variable like you could in older versions, which is why that error appears now but didn't used to.

the marlcham version should be deleted. /mixingsoftware/Chameleon/Version2015/raw_load_cham2 is the one to use.

I'll fix that.

Speaking of duplicate codes, there's a version of raw_load_chipod in /mixingsoftware/CTD_Chipod/mfiles/ that's now old compared to the version in /mixingsoftware/chipod/. I was hesitant to delete it though because you had made and organized that folder. However, the version of raw_load_chipod.m in /mixingsoftware/CTD_Chipod/mfiles/ crashes when trying to open newer chipod files (say from TAO14_140). Should I delete the old version?

It's really good to have someone else helping to debug mixingsoftware, because that's the way we'll see more problems...

**Comment by andypicke** [comment]
Yes, we should be able to remove the raw_load_chipod in /CTD_Chipod/, as
long as the new version still works for the older files. Go ahead and
remove it, and if it doesn't work i'll let you know :)

On Wed, Jun 29, 2016 at 4:43 PM, sallywarner notifications@github.com
wrote:

> okay. I think that makes sense. The marlcham version worked with old
> matlabs, but in 2016a (and maybe 2015a) you can't write over a global
> variable like you could in older versions, which is why that error appears
> now but didn't used to.
> 
> the marlcham version should be deleted.
> /mixingsoftware/Chameleon/Version2015/raw_load_cham2 is the one to use.
> 
> I'll fix that.
> 
> Speaking of duplicate codes, there's a version of raw_load_chipod in
> /mixingsoftware/CTD_Chipod/mfiles/ that's now old compared to the version
> in /mixingsoftware/chipod/. I was hesitant to delete it though because you
> had made and organized that folder. However, the version of
> raw_load_chipod.m in /mixingsoftware/CTD_Chipod/mfiles/ crashes when trying
> to open newer chipod files (say from TAO14_140). Should I delete the old
> version?
> 
> It's really good to have someone else helping to debug mixingsoftware,
> because that's the way we'll see more problems...
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/OceanMixingGroup/mixingsoftware/issues/2#issuecomment-229511385,
> or mute the thread
> https://github.com/notifications/unsubscribe/AG6El4Lyufv9JxFAt3_HaG17l52eQlRoks5qQvUWgaJpZM4JBnZ-
> .

## 

Andy Pickering
Postdoctoral Research Associate
Oregon State University
College of Earth, Ocean, and Atmospheric Sciences
978-764-3009
apickering@coas.oregonstate.edu
andypicke@gmail.com
Github https://github.com/andypicke
